### PR TITLE
FIX #8916 Add support to filter on zone-id for ec2 describe-availabili…

### LINKED ISSUE #8916: bug: ec2 describe-availability-zones with --zone-ids returns all zones
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Similar to [bug: ec2 describe-availability-zones with --zone-name returns all zones](https://github.com/localstack/localstack/issues/4715) but with `--zone-ids`.

```
aws --endpoint http://localhost:4566 ec2 describe-availability-zones
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1a",
            "ZoneId": "euw1-az3",
            "ZoneType": "availability-zone"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1b",
            "ZoneId": "euw1-az1",
            "ZoneType": "availability-zone"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1c",
            "ZoneId": "euw1-az2",
            "ZoneType": "availability-zone"
        }
    ]
}
```

```
aws --endpoint http://localhost:4566 ec2 describe-availability-zones --zone-ids euw1-az2
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1a",
            "ZoneId": "euw1-az3",
            "ZoneType": "availability-zone"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1b",
            "ZoneId": "euw1-az1",
            "ZoneType": "availability-zone"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1c",
            "ZoneId": "euw1-az2",
            "ZoneType": "availability-zone"
        }
    ]
}
```

### Expected Behavior

```
aws --endpoint http://localhost:4566 ec2 describe-availability-zones --zone-ids euw1-az2
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1c",
            "ZoneId": "euw1-az2",
            "ZoneType": "availability-zone"
        }
    ]
}
```
Similar to real AWS:
```
aws ec2 describe-availability-zones --zone-ids euw1-az2
{
    "AvailabilityZones": [
        {
            "State": "available",
            "OptInStatus": "opt-in-not-required",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1c",
            "ZoneId": "euw1-az2",
            "GroupName": "eu-west-1",
            "NetworkBorderGroup": "eu-west-1",
            "ZoneType": "availability-zone"
        }
    ]
}
```

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    awslocal s3 mb s3://mybucket

### Environment

```markdown
- OS:Ubuntu 22.04
- LocalStack: 2.2.0
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ashish1500616**
Fixes Issue #8916 

Label : https://github.com/localstack/localstack/labels/semver%3A%20patch
**Enhancements:**
- Adds support for filtering availability zone on zone-id and minor refactoring.

**Changes:**
- Use `moto.utilities.utils.filter_resources()` to filter zones based on multiple attributes, such as zone-name or zone-id.

### Before
```BASH
aws --endpoint http://localhost:4566 ec2 describe-availability-zones --zone-ids euw1-az2
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1a",
            "ZoneId": "euw1-az3",
            "ZoneType": "availability-zone"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1b",
            "ZoneId": "euw1-az1",
            "ZoneType": "availability-zone"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "eu-west-1",
            "ZoneName": "eu-west-1c",
            "ZoneId": "euw1-az2",
            "ZoneType": "availability-zone"
        }
    ]
}
```

```BASH
~ » awslocal ec2 describe-availability-zones --zone-name us-east-1f --zone-id use1-az5
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1f",
            "ZoneId": "use1-az5"
        }
    ]
}
~ » awslocal ec2 describe-availability-zones --zone-name us-east-1f
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1f",
            "ZoneId": "use1-az5"
        }
    ]
}
~ » awslocal ec2 describe-availability-zones --zone-id use1-az5
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1f",
            "ZoneId": "use1-az5"
        }
    ]
}
~ » awslocal ec2 describe-availability-zones --zone-id use1-az55
{
    "AvailabilityZones": []
}
~ » awslocal ec2 describe-availability-zones --zone-name us-east-1ff
{
    "AvailabilityZones": []
}
~ » awslocal ec2 describe-availability-zones --zone-name us-east                                                                                                    
{
    "AvailabilityZones": []
}
~ » awslocal ec2 describe-availability-zones --zone-name us-east-1e --zone-id use1-a                                                                                      
{
    "AvailabilityZones": []
}
~ » awslocal ec2 describe-availability-zones --zone-ids '["use1-az4","use1-az5"]'
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1d",
            "ZoneId": "use1-az4"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1f",
            "ZoneId": "use1-az5"
        }
    ]
}
awslocal ec2 describe-availability-zones --zone-names '["us-east-1e","us-east-1f"]'
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1e",
            "ZoneId": "use1-az3"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1f",
            "ZoneId": "use1-az5"
        }
    ]
}
awslocal ec2 describe-availability-zones --zone-ids '["use1-az5","use1-az3"]' --zone-names '["us-east-1e","us-east-1f"]'
{
    "AvailabilityZones": [
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1e",
            "ZoneId": "use1-az3"
        },
        {
            "State": "available",
            "Messages": [],
            "RegionName": "us-east-1",
            "ZoneName": "us-east-1f",
            "ZoneId": "use1-az5"
        }
    ]
}
```

### CODE REVIEW

File: localstack/services/ec2/provider.py [hunk 1]
```diff
@@ -97,25 +97,27 @@ def describe_availability_zones(
...
         zone_names = describe_availability_zones_request.get("ZoneNames")
-        if zone_names:
-            for zone in zone_names:
-                zone_detail = backend.get_zone_by_name(zone)
-                if zone_detail:
+        zone_ids = describe_availability_zones_request.get("ZoneIds")
+        if zone_names or zone_ids:
+            filters = {
+                "zone-name": zone_names if zone_names else None,
+                "zone-id": zone_ids if zone_ids else None,
```
Comment (viren-nadkarni) [inline localstack/services/ec2/provider.py:105]: These are redundant

```suggestion
                "zone-name": zone_names,
                "zone-id": zone_ids,
```

**Review by viren-nadkarni — CHANGES_REQUESTED** [review]
Thank you for contributing to LocalStack!

Just one nitpick and we can merge this

File: localstack/services/ec2/provider.py (hunk 1 above)
Comment (ashish1500616) [reply localstack/services/ec2/provider.py]: Hey @viren-nadkarni ,

Thank for the suggestion. I have made the required change.

**Comment by viren-nadkarni** [comment]
@ashish1500616 Could you please sign the CLA too?

**Comment by ashish1500616** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ashish1500616** [comment]
> Could you please sign the CLA too?

**Review by viren-nadkarni — APPROVED** [review]
LGTM! Congrats on your first contribution to LocalStack!

**Pull request merged**